### ISSUE TITLE
Default environment written to wrong place

### ISSUE THREAD

**Issue by josephcsible** [haskell/cabal#7943]
**Describe the bug**
Cabal writes the environment file under a subdirectory of ~/.ghc/, which then goes unused because GHC looks for it under a subdirectory of ~/.local/share/ghc/.

**To Reproduce**

```
docker run -it haskell:9.2 bash  # optional; the problem also happens with a regular ghcup install
cabal v2-update
cabal v2-install --lib contravariant
ghci
import Data.Functor.Contravariant.Divisible
```

**Expected behavior**
The import should succeed, since the package it's in was just installed and added to the default environment.

**System information**
 - Operating system: Debian buster
 - `cabal` 3.6.2.0, `ghc` 9.2.1

**Additional context**
This problem doesn't occur with older versions of GHC. Also, I discovered I can work around it by symlinking ~/.local/share/ghc/ to ~/.ghc/.

**Comment by gbaz** [comment]
cf https://github.com/haskell/cabal/issues/7819 -- its a bug in ghc which will hopefully be fixed in the next 9.2.x release, but arguably its unclear if cabal should also do something on top of that

**Comment by Mikolaj** [comment]
Good news, the GHC fix has just been backported and merged to GHC 9.2.2 (to be released).
